construct_binary_tree_from_preorder_and_inorder_traversal.java

```diff
@@ -0,0 +1,66 @@
+// Time Complexity : O(n), where n is the number of nodes in the tree (time requried to copy elements from inorder array to hashMap)
+// Space Complexity : O(n), O(n), where n is the number of nodes in the tree (space required for hashMap)
+// Did this code successfully run on Leetcode : Yes
+// Any problem you faced while coding this : I was trying to pass the preorderIndex as a local variable instead of global in recursion,

```

Comment:
preIndex is a primitive which is modified in every call. So it cannot be passed directly. If you do so, the variable won't retain expected value. If you want to pass it, you can create an array of size 1 and pass the array (pass by reference)